main.js

```diff
@@ -0,0 +1,241 @@
+    if (ingredient && ingredient.trim() !== "") {
+      ingredients.push({ name: ingredient.trim() });
+    }
+  }
+
+  const newRecipe = {
+    id: Date.now(),
+    name: title || "No title provided",
+    description: description || "No description available",
+    ingredients: ingredients.length > 0 ? ingredients : [{ name: "No ingredients available" }],
+    pictureUrl: pictureUrl || "",
+  };
+
+  recipes.push(newRecipe);
+
+  displayRecipes(recipes);
+
+  document.getElementById("recipe-form").reset();
+  resetIngredientFields();
+});
+
+function addIngredient() {
+  ingredientCount++;
+
+  const newIngredient = document.createElement("div");
+
+  const label = document.createElement("label");
+  label.setAttribute("for", `ingredient${ingredientCount}`);
+  label.textContent = `Ingredient ${ingredientCount}: `;
+
+  const input = document.createElement("input");
+  input.setAttribute("type", "text");
+  input.setAttribute("id", `ingredient${ingredientCount}`);
+  input.setAttribute("name", `ingredient${ingredientCount}`);
+
+  newIngredient.appendChild(label);
+  newIngredient.appendChild(input);
+
+  document.getElementById("ingredients-container").appendChild(newIngredient);
+}
+
+document.getElementById("add_more_ingredient").addEventListener("click", addIngredient);
+
+function resetIngredientFields() {
+  ingredientCount = 5;
+  document.getElementById("ingredients-container").innerHTML = `
+    <label for="ingredient1">Ingredient 1:</label>
+    <input type="text" id="ingredient1" name="ingredient1" required><br><br>
+
+    <label for="ingredient2">Ingredient 2:</label>
+    <input type="text" id="ingredient2" name="ingredient2" required><br><br>
+
+    <label for="ingredient3">Ingredient 3:</label>
+    <input type="text" id="ingredient3" name="ingredient3" required><br><br>
+
+    <label for="ingredient4">Ingredient 4:</label>
+    <input type="text" id="ingredient4" name="ingredient4" required><br><br>
+
+    <label for="ingredient5">Ingredient 5:</label>
+    <input type="text" id="ingredient5" name="ingredient5" required><br><br>
+  `;
+}
+
+async function searchRecipe() {
+  const searchInput = document.getElementById("searchInput").value.toLowerCase();
+  if (!recipes.length) {
+    await fetchRecipes();
+  }
+  const filteredRecipes = recipes.filter((recipe) =>
+    recipe.name.toLowerCase().includes(searchInput)
+  );
+
+  displayRecipes(filteredRecipes);
+}
+
+document.getElementById("search").addEventListener("submit", function (event) {
+  event.preventDefault();
+  searchRecipe();
+  document.getElementById("searchInput").value = "";
+});
+
+document.getElementById("search-recipe").addEventListener("click", searchRecipe);
+
+function filterByIngredientCount(maxCount) {
+  if (isNaN(maxCount) || maxCount < 1) {
+    alert("Please enter a valid number of ingredients.");
+    return;
+  }
+
+  const filteredRecipes = recipes.filter(recipe => recipe.ingredients.length <= maxCount);
+
+    filteredRecipes.sort((a, b) => {
```

Comment:
```suggestion
  filteredRecipes.sort((a, b) => {
```
Be careful with indentation here